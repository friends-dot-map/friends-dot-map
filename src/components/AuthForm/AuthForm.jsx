import React, { useState } from 'react';
import { Link } from 'react-router-dom';

export default function AuthForm({ handleAuth, isSigningUp }) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    handleAuth(email, password);
  };

  return (
    <>
      <h1 className="text-dark font-semibold font-cursive text-4xl bg-olive bg-opacity-20 w-full p-5 rounded-md text-center">
        <span className="text-teal">friends</span>.
        <span className="text-orange">map</span>(üìç)
      </h1>
      <div className="flex flex-col text-center items-center justify-evenly text-xl p-5 h-full">
        {isSigningUp ? (
          <p className="text-3xl font-semibold">Welcome!</p>
        ) : (
          <p className="text-3xl font-semibold">Welcome back!</p>
        )}
        <hr className="w-1/2 h-1 bg-dark opacity-30 border-1 rounded-md" />
        <form className="flex flex-col items-center w-3/4 space-y-5">
          <input
            aria-label="email"
            className="p-2 rounded-md"
            type="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            placeholder="email"
          />
          <input
            aria-label="password"
            className="p-2 rounded-md"
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            placeholder="password"
          />
          <button
            className="bg-teal text-white w-1/2 p-2 rounded-md"
            onClick={handleSubmit}
          >
            {isSigningUp ? 'Register' : 'Log In'}
          </button>
        </form>
        {isSigningUp ? (
          <>
            <p>
              Already have an account?{' '}
              <Link className="font-semibold block" to="/login">
                Log In
              </Link>
            </p>
          </>
        ) : (
          <>
            <p>
              Need to make an account?{' '}
              <Link className="font-semibold block" to="/register">
                Register
              </Link>
            </p>
          </>
        )}
        <p className="text-xl bg-white bg-opacity-70 p-5 rounded-md">
          A convenient travel companion to keep your group safe and organized
          when in unfamiliar places.
        </p>
        <div className="h-1/6"></div>
      </div>
    </>
  );
}
