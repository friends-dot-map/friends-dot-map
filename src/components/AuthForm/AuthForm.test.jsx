import AuthForm from "./AuthForm";
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom/";

test('auth form renders login page', () => {
    const { container } = render (
    <MemoryRouter>
        <AuthForm isSigningUp={true} />
    </MemoryRouter>
);
    
    screen.getByText(/already have an account\?/i)

    expect(container).toMatchSnapshot();
});

test('auth form renders register page', () => {
    const { container } = render (
    <MemoryRouter>
        <AuthForm isSigningUp={false} />
    </MemoryRouter>
);
    
    screen.getByText(/need to make an account\?/i)
    expect(container).toMatchSnapshot();
});