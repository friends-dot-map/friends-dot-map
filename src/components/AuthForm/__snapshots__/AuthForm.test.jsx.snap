// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth form renders login page 1`] = `
<div>
  <div>
    <form>
      <label>
        email:
        <input
          placeholder="please enter your email"
          type="email"
          value=""
        />
      </label>
      <label>
        password:
        <input
          placeholder="please enter your password"
          type="password"
          value=""
        />
      </label>
      <button>
        Register
      </button>
    </form>
    <p>
      a convenient travel companion to keep your group safe and organized when in unfamiliar places
    </p>
    <p>
      Already have an account? 
      <a
        href="/login"
      >
        Log In
      </a>
    </p>
  </div>
</div>
`;

exports[`auth form renders register page 1`] = `
<div>
  <div>
    <form>
      <label>
        email:
        <input
          placeholder="please enter your email"
          type="email"
          value=""
        />
      </label>
      <label>
        password:
        <input
          placeholder="please enter your password"
          type="password"
          value=""
        />
      </label>
      <button>
        Log In
      </button>
    </form>
    <p>
      a convenient travel companion to keep your group safe and organized when in unfamiliar places
    </p>
    <p>
      Need to make an account? 
      <a
        href="/register"
      >
        Register
      </a>
    </p>
  </div>
</div>
`;
